src/components/modals/ExcludeClassTimeFilters.tsx

```diff
@@ -0,0 +1,247 @@
+import { Add, DeleteForeverRounded, DeleteRounded, RemoveRounded } from "@mui/icons-material";
+import {
+    Box,
+    Button,
+    Divider,
+    FormControl,
+    FormGroup,
+    IconButton,
+    InputLabel,
+    MenuItem,
+    Select,
+    Stack,
+    Tooltip,
+    Typography,
+    useTheme,
+} from "@mui/material";
+import { TimePicker } from "@mui/x-date-pickers";
+import { DateTime } from "luxon";
+import React, { Dispatch, SetStateAction, useCallback, useEffect, useState } from "react";
+
+import { getCapitalizedWeekdays } from "@/lib/helpers/date";
+import { storeExcludedClassTimeFilters } from "@/lib/helpers/storage";
+import { ExcludeClassTimeFilter } from "@/types/chain";
+
+export default function ExcludeClassTimeFilters({
+    excludedClassTimeFilters,
+    setExcludedClassTimeFilters,
+}: {
+    excludedClassTimeFilters: ExcludeClassTimeFilter[];
+    setExcludedClassTimeFilters: Dispatch<SetStateAction<ExcludeClassTimeFilter[]>>;
+}) {
+    const theme = useTheme();
+
+    const weekdays = getCapitalizedWeekdays();
+    const [excludedClassTimeWeekday, setExcludedClassTimeWeekday] = useState<number>(1);
+    const [excludedClassTimeStartTime, setExcludedClassTimeStartTime] = useState<DateTime | null>(null);
+    const [excludedClassTimeEndTime, setExcludedClassTimeEndTime] = useState<DateTime | null>(null);
+    const [inputValid, setInputValid] = useState<boolean>(false);
+
+    const validateInput = useCallback(() => {
+        return (
+            excludedClassTimeWeekday >= 1 &&
+            excludedClassTimeWeekday <= 7 &&
+            excludedClassTimeStartTime?.isValid === true &&
+            excludedClassTimeEndTime?.isValid === true &&
+            excludedClassTimeStartTime < excludedClassTimeEndTime &&
+            !excludedClassTimeFilters.some(
+                (it) =>
+                    it.weekday === excludedClassTimeWeekday &&
+                    it.startHour === excludedClassTimeStartTime.hour &&
+                    it.startMinute === excludedClassTimeStartTime.minute &&
+                    it.endHour === excludedClassTimeEndTime.hour &&
+                    it.endMinute === excludedClassTimeEndTime.minute,
+            )
+        );
+    }, [excludedClassTimeFilters, excludedClassTimeWeekday, excludedClassTimeStartTime, excludedClassTimeEndTime]);
+
+    useEffect(() => {
+        setInputValid(validateInput());
+    }, [
+        validateInput,
+        excludedClassTimeFilters,
+        excludedClassTimeWeekday,
+        excludedClassTimeStartTime,
+        excludedClassTimeEndTime,
+    ]);
```

Comment:
How about a `useMemo` instead of `useState`+`useEffect`?

Reply:
I'm not familiar with `useMemo`, so will have to look into it... This was the "quick and dirty" way :v:

Reply:
Have not tested this, but might be as simple as just
```jsx
const inputValid = useMemo(() => validateInput() !== false, [validateInput])
```

Reply:
:white_check_mark: